### DISCUSSION TITLE
Product Personalization/Customization Fields

### DISCUSSION THREAD

**Discussion by SDG-Wes** [spree/spree#11246] [Ideas]
Much like how special instructions works for shipping, the need for customers to provide specific instructions for product customizations/personalizations is often needed.   Ideally, this field for user input would occur on the product order page just like choosing a variant (i.e. color, size)

While one field would be nice, the idea of having multiple entries would be ideal with the ability to limit the character count.  I.e. Name for engraving (25 char), title for engraving (30)

This is an advertised feature of Spree yet requires core modifications which would need to be maintained going forward and I'm sure is something that would be helpful to many people using Spree Commerce.

**Comment by damianlegawiec** [comment]
The next Spree release will include the Metadata feature which means storing/accessing custom data for Products, Orders, Customers, and other resources. This will be available via API.

**Reply by fthobe** [reply]
@damianlegawiec Hey, I couldn't find anything in the documentation, could you point me in the right direction?

**Reply by damianlegawiec** [reply]
Hi @fthobe 

We introduced `Metadata` starage concept some time ago - https://github.com/spree/spree/blob/main/core/app/models/concerns/spree/metadata.rb

Several models are using it already, you can check it in the code.
